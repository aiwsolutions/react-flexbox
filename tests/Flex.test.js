import React from 'react';
import renderer from 'react-test-renderer';
import { Flex } from '../dist';

test('<Flex> - basic settings', () => {
    const comp = renderer.create(
        <Flex
            style={{ backgroundColor: '#ff0000' }}
            m={1}
            p={1}
            column
        >
            Title
        </Flex>
    );
    expect(comp.toJSON()).toMatchSnapshot();
});

test('<Flex> - className should be passed correctly', () => {
    const comp = renderer.create(
        <Flex
            className="test"
            inline
        />
    );
    expect(comp.toJSON()).toMatchSnapshot();
    expect(comp.root.findByType('div').props.className).toEqual('inline-flex test');
});

test('<Flex> - update on property changes', () => {
    const comp = renderer.create(
        <Flex
            align="center"
        />
    );
    const shouldComponentUpdate = comp.getInstance().shouldComponentUpdate.bind(comp.getInstance());
    expect(shouldComponentUpdate({ align: 'flex-start' })).toBeTruthy();
});

test('<Flex> - update on layout changes', () => {
    const comp = renderer.create(
        <Flex
            m={1}
        />
    );
    const shouldComponentUpdate = comp.getInstance().shouldComponentUpdate.bind(comp.getInstance());
    expect(shouldComponentUpdate({ m: 2 })).toBeTruthy();
});

test('<Flex> - should not update if nothing changes', () => {
    const comp = renderer.create(
        <Flex
            justify="center"
            style={{ backgroundColor: '#fff' }}
            p={1}
        >
            Title
        </Flex>
    );
    const shouldComponentUpdate = comp.getInstance().shouldComponentUpdate.bind(comp.getInstance());
    expect(shouldComponentUpdate(comp.getInstance().props)).toBeFalsy();
});
