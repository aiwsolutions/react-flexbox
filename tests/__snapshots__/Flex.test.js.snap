// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Flex> - basic settings 1`] = `
<div
  className="flex"
  style={
    Object {
      "WebkitBoxSizing": "border-box",
      "WebkitFlexDirection": "column",
      "backgroundColor": "#ff0000",
      "boxSizing": "border-box",
      "flexDirection": "column",
      "margin": 8,
      "padding": 8,
    }
  }
>
  Title
</div>
`;

exports[`<Flex> - className should be passed correctly 1`] = `
<div
  className="inline-flex test"
  style={
    Object {
      "WebkitBoxSizing": "border-box",
      "WebkitFlexDirection": "row",
      "boxSizing": "border-box",
      "flexDirection": "row",
    }
  }
/>
`;
